src/main/java/com/allenai/ml/linalg/SparseVector.java

```diff
@@ -87,5 +89,5 @@ public double dotProduct(Vector other) {
           result += val * other.at(idx);
```

Comment:
perf question: you said this was backed by long[]  + double[], right?  So this dot product is iterating over the smaller vector (array iteration, fast), but doing lookups into the other one - so log(numStoredEntries) binary search per entry, for a total cost of smaller.numStoredEntries() * log(larger.numStoredEntries()), right?  Should be able to be done linearly if you iterate in parallel, no?